src/rest-server/src/config/v2/user.js

```diff
@@ -77,14 +77,102 @@ const userCreateInputSchema = Joi.object().keys({
     .default(),
 });
 
+// define the input schema for the 'update user' api in basic mode for admin user
+const basicAdminUserUpdateInputSchema = Joi.object().keys({
+  patch: Joi.boolean()
+    .default(false),
+  data: Joi.object().keys({
+    username: Joi.string()
+      .regex(/^[\w.-]+$/, 'username')
+      .required(),
+    email: Joi.string()
+      .email(),
+    virtualCluster: Joi.array()
+      .items(Joi.string()),
+    admin: Joi.boolean(),
+    password: Joi.string()
+      .min(6),
+    extension: Joi.object()
+      .pattern(/\w+/, Joi.required()),
+  })
+    .when('patch', {
+        is: true,
+        then: Joi.object({
+          email: Joi.empty(null),
+          virtualCluster: Joi.empty(null),
+          admin: Joi.empty(null),
+          password: Joi.empty(null),
+          extension: Joi.empty(null),
+        }),
+        otherwise: Joi.object({
+          email: Joi.required(),
+          virtualCluster: Joi.required(),
+          admin: Joi.required(),
+          password: Joi.required(),
+          extension: Joi.required(),
+        }),
+      }
+    ),
+});
+
+// define the input schema for the 'update user' api in basic mode for all user
+const basicUserUpdateInputSchema = Joi.object().keys({
+  patch: Joi.boolean()
+    .default(false),
+  data: Joi.object().keys({
+    username: Joi.string()
+      .regex(/^[\w.-]+$/, 'username')
+      .required(),
+    email: Joi.string()
+      .email(),
+    newPassword: Joi.string()
+      .min(6),
+    oldPassword: Joi.string()
+      .min(6)
+      .when('newPassword', {
+        is: Joi.exist(),
+        then: Joi.required(),
+      }),
+  })
+    .when('patch', {
+        is: true,
+        then: Joi.object({
+          email: Joi.empty(null),
+          newPassword: Joi.empty(null),
+        }),
+        otherwise: Joi.object({
+          email: Joi.required(),
+          newPassword: Joi.required(),
+        }),
+      }
+    ),
+});
+
+// define the input schema for the 'update user' api in oidc mode
+const oidcUserUpdateInputSchema = Joi.object().keys({
```

Comment:
actually this is for `admin user` only, right? if so, let's rename it to `oidcAdminUserUpdateInputSchema`